### PR TITLE
MHV-50152 Added app-level feature toggle for MR

### PR DESCRIPTION
**Pull request by mmoyer-va**
## Summary

Added a new feature toggle for MR. When disabled, the user is unable to browse to MR, instead being redirected back to the va.gov home page.

Backend component: https://github.com/department-of-veterans-affairs/vets-api/pull/14084

### [MHV-50152](https://jira.devops.va.gov/browse/MHV-50152) - Change production flag to true for MR for UAT ###

> In order for UAT to take place in a production environment, MR must be visible in production.
> 
> It is not linked from any other part of the VA.gov website, so will be "dark deployed." Only those participating in UAT will know to browse there.

## Testing done

New unit tests to test various feature flag states (on, off, not-yet-loaded).

## Screenshots

Unable to take screenshot. If the feature flag is disabled, the page instantly reloads to va.gov home.

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [ ] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [x] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [x] Did you login to a local build and verify all authenticated routes work as expected with a test user

### CODE REVIEW

**Review by mattwrightva — APPROVED** [review]
Looks good!

File: src/applications/mhv/medical-records/tests/containers/App.unit.spec.jsx [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+
+    it('feature flag set to false', () => {
+      const screen = renderWithStoreAndRouter(
+        <App>
+          <LandingPage />
+        </App>,
+        {
+          initialState: {
+            featureToggles: {
+              // eslint-disable-next-line camelcase
```
[the remark this answers is not in the document]
Comment (mmoyer-va) [reply src/applications/mhv/medical-records/tests/containers/App.unit.spec.jsx:89]: Setting the state in the unit test requires not using camelcase.

File: src/applications/mhv/medical-records/tests/containers/App.unit.spec.jsx [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+
+    it('feature flag set to true', () => {
+      const screen = renderWithStoreAndRouter(
+        <App>
+          <LandingPage />
+        </App>,
+        {
+          initialState: {
+            featureToggles: {
+              // eslint-disable-next-line camelcase
```
[the remark this answers is not in the document]
Comment (mmoyer-va) [reply src/applications/mhv/medical-records/tests/containers/App.unit.spec.jsx:120]: Setting the state in the unit test requires not using camelcase.

File: src/applications/mhv/medical-records/tests/fixtures/mocks/ResizeObserver.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
+/* eslint-disable class-methods-use-this */
```
[the remark this answers is not in the document]
Comment (mmoyer-va) [reply src/applications/mhv/medical-records/tests/fixtures/mocks/ResizeObserver.js:1]: Mocked up a class for tests. Its methods have nothing internally, so I cannot make use of the "this" keyword there.

**Review by ataker — APPROVED** [review]

**Pull request merged**